packages/presentation/lib/src/game_bloc/game_bloc.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:domain/domain.dart';
+import 'package:presentation/src/constants/init_values.dart';
+import 'package:presentation/src/game_bloc/bloc.dart';
+import 'package:presentation/src/game_bloc/bloc_tile.dart';
+import 'package:presentation/src/game_bloc/events.dart';
+
+abstract class GameBloc extends Bloc {
+  factory GameBloc(
+    GenerateGuessNumberUseCase generateGuessNumber,
+    MakeAttemptUseCase makeAttempt,
+  ) =>
+      GameBlocImpl(makeAttempt, generateGuessNumber);
+
+  void add(BlocEvent event);
+}
+
+class GameBlocImpl extends BlocImpl implements GameBloc {
+  final GenerateGuessNumberUseCase _generateGuessNumberUseCase;
+  final MakeAttemptUseCase _makeAttemptUseCase;
+  int _suggestedNumber = blocTileInitValue;
+
+  GameBlocImpl(this._makeAttemptUseCase, this._generateGuessNumberUseCase) {
+    eventStream.listen(
+      (event) {
+        if (event is MakeAttemptEvent)
+          _makeAttempt();
+        else if (event is NewGameEvent)
+          _newGame();
+        else if (event is SetSuggestedNumberEvent)
+          _setSuggestedNumber(event.number);
+      },
+    );
```

Comment:
let's call bloc methods directly instead of passing events to it (bloc -> cubit)